pyanaconda/packaging/yumpayload.py

```diff
@@ -1611,3 +1637,69 @@ def postInstall(self):
         # Make sure yum is really done and gone and lets go of the yum.log
         self._yum.close()
         del self._yum
+
+class RepoMDMetaHash(object):
+    """Class that holds hash of a repomd.xml file content from a repository.
+    This class can test availability of this repository by comparing hashes.
+    """
+    def __init__(self, payload, repo):
+        self._repoId = repo.id
+        self._method = payload.data.method.method
+        self._urls = repo.urls
+        self._repomd_hash = ""
+
+    @property
+    def repoMDHash(self):
+        """Return MD5 hash of the repomd.xml file stored."""
+        return self._repomd_hash
+
+    @property
+    def id(self):
+        """Name of the repository."""
+        return self._repoId
+
+    def storeRepoMDHash(self):
+        """Download and store hash of the repomd.xml file content."""
+        repomd = self._downloadRepoMD(self._method)
+        self._repomd_hash = self._calculateHash(repomd)
+
+    def verifyRepoMD(self):
+        """Download and compare with stored repomd.xml file."""
+        new_repomd = self._downloadRepoMD(self._method)
+        new_repomd_hash = self._calculateHash(new_repomd)
+        return new_repomd_hash == self._repomd_hash
+
+    def _calculateHash(self, message):
+        m = hashlib.md5()
+        m.update(message)
```

Comment:
First I used `input` here but then pylint complain about redefining `input`... so I changed it to a neutral `message`. Do you have better name for this?